### DISCUSSION TITLE
Model viewer on click get the node/material name.

### DISCUSSION THREAD

**Discussion by alekko** [google/model-viewer#2490] [Q&A]
Hi!

Is it possible from the onClick event to detect on what material/node did the client click so one can get the material/node name and then add the ability to interact with that material?

Thanks in advance!

**Comment by elalish** [comment]
Not yet, but we're working on it. @timmmeh

**Reply by softmarshmallow** [reply]
Would love to play with this new feature!

**Reply by softmarshmallow** [reply]
Related: https://github.com/google/model-viewer/issues/2572

**Reply by sylvainlec** [reply]
I would love to see this feature as well!
@alekko I am contourning this by setting one material per node with custom name in the source glb file. I do not care about the originals colors in my case, as I need to rebuild them from scratch depending on user input on the frontend.
Like that, I can use the materialFromPoint() fonction and it returns me the material with it's custom name corresponding to the component name.
![image](https://user-images.githubusercontent.com/54276247/141358731-6fb42cad-b60d-4e34-9918-d83a770deb3b.png)
